test/e2e/linked_clone_test.go

```diff
@@ -0,0 +1,136 @@
+	var (
+		namespace *corev1.Namespace
+	)
+
+	BeforeEach(func() {
+		Expect(bootstrapClusterProxy).NotTo(BeNil(), "BootstrapClusterProxy can't be nil")
+		namespace = setupSpecNamespace("linked-clone-e2e")
+	})
+
+	AfterEach(func() {
+		cleanupSpecNamespace(namespace)
+	})
+
+	It("should create a cluster by using linked clone to create nodes", func() {
+		VerifyLinkedClone(ctx, LinkedCloneSpecInput{
+			Namespace: namespace,
+			InfraClients: InfraClients{
+				Client:     vsphereClient,
+				RestClient: restClient,
+				Finder:     vsphereFinder,
+			},
+			Global: GlobalInput{
+				BootstrapClusterProxy: bootstrapClusterProxy,
+				ClusterctlConfigPath:  clusterctlConfigPath,
+				E2EConfig:             e2eConfig,
+				ArtifactFolder:        artifactFolder,
+			},
+		})
+	})
+})
+
+func VerifyLinkedClone(ctx context.Context, input LinkedCloneSpecInput) {
+	var (
+		specName         = "linked-clone"
+		namespace        = input.Namespace
+		clusterResources = new(clusterctl.ApplyClusterTemplateAndWaitResult)
+	)
+
+	clusterName := fmt.Sprintf("%s-%s", specName, util.RandomString(6))
+	Expect(namespace).NotTo(BeNil())
+
+	Byf("creating a workload cluster %s", clusterName)
+	configCluster := defaultConfigCluster(clusterName, namespace.Name, "", 1, 1, input.Global)
+
+	clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+		ClusterProxy:                 input.Global.BootstrapClusterProxy,
+		ConfigCluster:                configCluster,
+		WaitForClusterIntervals:      input.Global.E2EConfig.GetIntervals("", "wait-cluster"),
+		WaitForControlPlaneIntervals: input.Global.E2EConfig.GetIntervals("", "wait-control-plane"),
+		WaitForMachineDeployments:    input.Global.E2EConfig.GetIntervals("", "wait-worker-nodes"),
+	}, clusterResources)
+
+	By("verifying linked clone")
+	vsphereVMs := FetchVsphereVMsForCluster(ctx, input.Global.BootstrapClusterProxy, clusterName, namespace.Name)
+	Expect(verifyDiskLayoutOfVMs(ctx, input.Finder, vsphereVMs)).To(Succeed())
+}
+
+func verifyDiskLayoutOfVMs(ctx context.Context, finder *find.Finder, vms []infrav1.VSphereVM) error {
+	for _, vm := range vms {
+		vmObj, err := finder.VirtualMachine(ctx, vm.Name)
+		if err != nil {
+			return err
+		}
+
+		var vmMo mo.VirtualMachine
+		err = vmObj.Properties(ctx, vmObj.Reference(), []string{"layoutEx"}, &vmMo)
+		if err != nil {
+			return err
+		}
+
+		for _, disk := range vmMo.LayoutEx.Disk {
+			if len(disk.Chain) < 2 {
+				return errors.New("Disk file is not chained")
+			}
+		}
+	}
+	return nil
+}
+
+func FetchVsphereVMsForCluster(ctx context.Context, bootstrapClusterProxy framework.ClusterProxy, clusterName, ns string) []infrav1.VSphereVM {
+	vms := &infrav1.VSphereVMList{}
+	err := bootstrapClusterProxy.GetClient().List(
+		ctx,
+		vms,
+		client.InNamespace(ns),
+		client.MatchingLabels{
+			clusterv1.ClusterLabelName: clusterName,
+		})
+	Expect(err).ToNot(HaveOccurred())
+
+	return vms.Items
+}
```

Comment:
Use the `getVSphereVMsForCluster` defined in the `storage_policy_test.go` file instead.